### ISSUE TITLE
Cannot check private repositories

### ISSUE THREAD

**Issue by danielhaap83** [EmbarkStudios/cargo-deny-action#53]
**Describe the bug**
When my cargo project has a private git repository defined, cargo deny action will fail. Please add possibility to provide a personal access token for git usage.

**To Reproduce**
1. Create cargo crate which references a private git repository
2. Configure deny as usual (defaults are enough)
3. Run 'cargo deny --log-level warn --all-features check bans licenses sources'
4. Fails with:
```sh
[ERROR] `cargo metadata` exited with an error:     Updating crates.io index
    Updating git repository `[https://github.com/private-repo`](https://github.com/private-repo%60)
error: failed to get `private-repo` as a dependency of package `...`
Caused by:
  failed to load source for dependency `private-repo`
...
```

My action.yaml
```yaml
deny:
    name: Deny
    runs-on: ubuntu-latest
    strategy:
      matrix:
        checks:
          - advisories
          - bans licenses sources
    # Prevent sudden announcement of a new advisory from failing ci:
    continue-on-error: ${{ matrix.checks == 'advisories' }}
    steps:
      # checkout this repository
      - name: Checkout
        uses: actions/checkout@v2
      # checkout my private repository - just to show that it works
      - name: Checkout
        uses: actions/checkout@v2
        with:
          repository: private-repo
          path: tmp
          ref: master
          token: ${{ secrets.USER_PAT }}
      - name: Deny
        uses: EmbarkStudios/cargo-deny-action@v1
        with:
          command: check ${{ matrix.checks }}
```

**Expected behavior**
deny should run and check as "normal" and provide its results. It woks also locally using gits credential-helper=store. Please support option to pass token/credentials to cargo-deny-action.

**Comment by Jake-Shadle** [comment]
IMO it would be cleaner to do a `cargo fetch` with whatever secrets and setup you normally do before running the deny action, I really don't want to add support in the action itself for niche scenarios like this.

**Comment by danielhaap83** [comment]
Nice idea but the fetched sources are also not part of executed docker container. I also tried to use the `--disable-fetch` flag. Without success.

**Comment by Jake-Shadle** [comment]
Ok, well, I don't know how one normally sets up cargo to clone private repos so I'd accept a PR that does it in whatever normal way works for users I guess.

**Comment by danielhaap83** [comment]
A simple example is here: https://github.com/de-vri-es/setup-git-credentials. This writes a gitconfig and enables the git credential store.

In addition it redirects all SSH URLs to https once to make the store working also for SSH Urls. I don't think this is needed for cargo.

I will try to setup something simple.

**Comment by Jake-Shadle** [comment]
Resolved by #54
